provider/gateway/client.go

```diff
@@ -38,22 +44,66 @@ type ServiceLogMessage struct {
 }
 
 type ServiceLogs struct {
-	Stream <-chan ServiceLogMessage
+	Stream  <-chan ServiceLogMessage
+	OnClose <-chan string
 }
 
+type ClientOptions func(Client)
+
 // NewClient returns a new Client
-func NewClient() Client {
-	return &client{
-		hclient: http.DefaultClient,
+func NewClient(cctx cclient.Context, addr sdk.Address, certs []tls.Certificate) (Client, error) {
```

Comment:
Is it worth checking in the code to see if someone passes `certs` with `len(certs) == 0`? No one should, but it is a common problem.

Reply:
`certs` can be empty if there is no need in authenticated request, for example provider status.
point to update the doc